data/pigui/modules/radar.lua

```diff
@@ -95,144 +164,212 @@ local function display2DRadar(cntr, size)
-				Event.Queue('onChangeMFD', 'scanner')
+				toggle_radar = true
 			end
 		end)
-		if shouldDisplay2DRadar then
-			display2DRadar(cntr, size)
-		else
-			display3DRadar(cntr, Vector2(ui.reticuleCircleRadius * 1.8, size * 2))
+
+		if ui.isMouseClicked(1) then
+			click_on_radar = true
+		end
+		if not toggle_radar and click_on_radar and ui.isMouseReleased(1) then
+			ui.openPopup("radarselector")
+		end
+		-- TODO: figure out how to "capture" the mouse wheel to prevent
+		-- the game engine from using it to also zoom the viewport
+		if zoom == 0 then
+			zoom = ui.getMouseWheel()
 		end
 	end
-end
 
-Event.Register('onChangeMFD', function(selected)
-	shouldDisplay2DRadar = selected == "radar";
-end)
+	-- Update the zoom level
+	if zoom > 0 then
+		instrument:zoomIn()
+	elseif zoom < 0 then
+		instrument:zoomOut()
+	end
+
+	-- Draw the actual radar - this can't be in a window or the 3D scanner background doesn't render
+	instrument:draw(center)
+
+	-- Draw the radar buttons and info - this needs to be in a window, otherwise the buttons don't work.
+	local window_width = ui.reticuleCircleRadius * 1.8
+	local window_height = radar2d.size / 3.5
+	local window_pos = Vector2(center.x - window_width / 2, center.y + radar2d.size - window_height - SCREEN_BORDER)
+	local windowFlags = ui.WindowFlags {"NoTitleBar", "NoResize", "NoFocusOnAppearing", "NoBringToFrontOnFocus", "NoSavedSettings"}
+	ui.setNextWindowPos(window_pos, "Always")
+	ui.setNextWindowPadding(Vector2(0))
+	ui.setNextWindowSize(Vector2(window_width, window_height), "Always")
+	ui.window("radar_buttons", windowFlags, function()
+		-- Draw radar mode toggle button
+		local icon = shouldDisplay2DRadar and radar3d.icon or radar2d.icon
+		local clicked = ui.mainMenuButton(icon, lui.HUD_RADAR_TOGGLE_MODE, false, Vector2(window_height))
+		if toggle_radar or clicked then
+			shouldDisplay2DRadar = not shouldDisplay2DRadar
+		end
+
+		-- Draw zoom mode indicator
+		if not shouldDisplay2DRadar then
+			local button_size = window_height / 1.5
+			local tt = radar3d.auto_zoom and lui.HUD_RADAR_ZOOM_MODE_AUTOMATIC or lui.HUD_RADAR_ZOOM_MODE_MANUAL
+			ui.sameLine()
+			ui.addCursorPos(Vector2(0, window_height - button_size))
+			icon = instrument:isAutoZoom() and icons.radar_automatic or icons.radar_manual
+			ui.mainMenuButton(icon, tt, ui.theme.buttonColors.disabled, Vector2(button_size))
+		end
+
+		-- Draw radar range
+		local distance = ui.Format.Distance(instrument:getZoom())
+		local textpos = ui.getWindowPos() + Vector2(window_width, window_height)
+		ui.addStyledText(textpos, ui.anchor.right, ui.anchor.bottom, distance, colors.frame, pionillium.small, lui.HUD_RADAR_DISTANCE, colors.lightBlackBackground)
+	end) -- window
+end -- function displayRadar()
 
 -- reset radar to default at game end
-Event.Register("onGameEnd", function() shouldDisplay2DRadar = false end)
+Event.Register("onGameEnd", function()
+	shouldDisplay2DRadar = false,
+	radar2d:resetZoom(),
+	radar3d:resetZoom()
```

Comment:
Left-over commas here turn this into a three-value assignment to a single variable rather than three statements.

Reply:
Fixed in 4f2069f05c56d9475e530db9feb17fb57059d159